src/toil/server/wes/toil_backend.py

```diff
@@ -0,0 +1,335 @@
+        temp_dir = job.exec_dir
+        try:
+            _, request = self.collect_attachments(run_id, temp_dir=temp_dir)
+        except ValueError:
+            job.clean_up()
+            raise
+
+        wf_type = request["workflow_type"].lower().strip()
+        version = request["workflow_type_version"]
+
+        # validate workflow request
+        supported_versions = self.supported_versions.get(wf_type, None)
+        if not supported_versions:
+            job.clean_up()
+            raise RuntimeError(f"workflow_type '{wf_type}' is not supported.")
+        if version not in supported_versions:
+            job.clean_up()
+            raise RuntimeError("workflow_type '{}' requires 'workflow_type_version' to be one of '{}'.  Got '{}'"
+                               "instead.".format(wf_type, str(supported_versions), version))
+
+        job.set_state("QUEUED")
+
+        command = job.get_run_command(request=request, options=self.options)
+
+        # TODO: Flask recommends using a task queue (like Celery or rq) to run tasks like these, since this way
+        #  will likely leave behind zombie child processes until we read the return code of the process.
+        # https://flask.palletsprojects.com/en/2.0.x/patterns/celery/
+        # https://python-rq.org/
+
+        with open(os.path.join(job.work_dir, "runner.log"), "w") as log:
+            process = subprocess.Popen(command, stdout=log, stderr=log, cwd=job.work_dir)
+
+        with open(os.path.join(job.work_dir, "runner.pid"), "w") as f:
+            f.write(str(process.pid))
+
+        self.processes[run_id] = process
+        logger.info(f"Spawned child process ({process.pid}) to run the requested workflow.")
+
+        return {
+            "run_id": run_id
+        }
+
+    @handle_errors
+    def get_run_log(self, run_id: str) -> Dict[str, Any]:
+        """ Get detailed info about a workflow run."""
+        run = self._get_run(run_id, should_exists=True)
+        state = run.get_state()
+
+        request = json.loads(run.fetch("request.json", "{}"))
+
+        cmd = run.fetch("cmd", "").split("\n")
+        start_time = run.fetch("start_time")
+        end_time = run.fetch("end_time")
+
+        # TODO: stdout and stderr should be a URL that points to the output file, not the actual contents.
+        stdout = run.fetch("stdout")
+        stderr = run.fetch("stderr")
+        exit_code = run.fetch("exit_code")
+
+        output_obj = {}
+        if state == "COMPLETE":
+            output_obj = run.get_output_files()
+
+        return {
+            "run_id": run_id,
+            "request": request,
+            "state": state,
+            "run_log": {
+                "cmd": cmd,
+                "start_time": start_time,
+                "end_time": end_time,
+                "stdout": stdout,
+                "stderr": stderr,
+                "exit_code": int(exit_code) if exit_code is not None else None,
+            },
+            "task_logs": [],
```

Comment:
I believe including task logs inside of the main logs endpoint is probably clunky and bad.  I'll suggest that we don't touch this now and leave this as `[]`, but propose a change to the WES API itself later to have a separate endpoint that accepts a task_id and run_id for specific logs and have this list include only the task_ids, and maybe success or failure.

Reply:
The task log entries just have links to the logs, not the full text. I feel like we should implement WES as specified now, rather than just waiting around until we can talk them into changing it.

Reply:
@adamnovak That makes sense.  I should have gone over the spec in more detail.